src/core/session.py

```diff
@@ -236,3 +247,62 @@ def sessionDirectory(project: QgsProject) -> str:
                 return uri.split("|")[0].replace(DB_NAME, "")
 
         return ""
+
+    @staticmethod
+    def merge(sessionADir, sessionBDir, sessionOutputDir):
+        # open input session
+        sessionA = SammoSession()
+        sessionA.init(sessionADir, load=False)
+
+        sessionB = SammoSession()
+        sessionB.init(sessionBDir, load=False)
+
+        # copy wav files to output session
+        for session in [sessionA, sessionB]:
+            for wav in session.wavFiles:
+                copy(wav, sessionOutputDir)
+
+        # create output session
+        sessionOutput = SammoSession()
+        sessionOutput.init(sessionOutputDir, load=False)
+
+        # copy features from dynamic layers
+        request = QgsFeatureRequest(QgsExpression('"copy" = 0'))
+
+        dynamicLayers = [
+            "environmentLayer",
+            "sightingsLayer",
+            "gpsLayer",
+            "followersLayer",
+        ]
+        for layer in dynamicLayers:
+            out = getattr(sessionOutput, layer)
+            out.startEditing()
+
+            newFid = 0
+            lastFeature = SammoDataBase.lastFeature(out)
+            if lastFeature:
+                newFid = lastFeature["fid"] + 1
+
+            for vl in [getattr(sessionA, layer), getattr(sessionB, layer)]:
+                for feature in vl.getFeatures(request):
+                    feature["fid"] = newFid
+                    feature["copy"] = 1
```

Comment:
@pblottiere Hmmm I'm not sure about the purpose of this attribute. If I do 2 sessions A and B, and I merge both into C, then I can't merge C with an eventual D session to a E session, because all features in C (with copy=1) will be not copy, no ?

Reply:
Yes indeed.